packages/playwright-test/src/dispatcher.ts

```diff
@@ -17,18 +17,21 @@
 import child_process from 'child_process';
 import path from 'path';
 import { EventEmitter } from 'events';
-import type { RunPayload, TestBeginPayload, TestEndPayload, DonePayload, TestOutputPayload, WorkerInitParams, StepBeginPayload, StepEndPayload, SerializedLoaderData, TeardownErrorsPayload } from './ipc';
+import type { RunPayload, TestBeginPayload, TestEndPayload, DonePayload, TestOutputPayload, WorkerInitParams, StepBeginPayload, StepEndPayload, SerializedLoaderData, TeardownErrorsPayload, TestServerTestResolvedPayload } from './ipc';
 import type { TestResult, Reporter, TestStep, TestError } from '../types/testReporter';
-import type { Suite, TestCase } from './test';
+import type { Suite } from './test';
 import type { Loader } from './loader';
+import { TestCase } from './test';
 import { ManualPromise } from 'playwright-core/lib/utils/manualPromise';
+import { TestTypeImpl } from './testType';
 
 export type TestGroup = {
   workerHash: string;
   requireFile: string;
   repeatEachIndex: number;
   projectId: string;
   tests: TestCase[];
+  testServerTestLine?: number;
```

Comment:
You need `requireFile` + `uiLocation`.

Reply:
I'd like to keep it as is as a reminder to the limitation.